2005-03-30  Jeff Johnston  <jjohnstn@redhat.com>

	* bfd/elfcode.h: Change code to allow for idiosyncrasies of
	ia64 vsyscall page.

Index: gdb-6.5/bfd/elfcode.h
===================================================================
--- gdb-6.5.orig/bfd/elfcode.h	2006-05-11 05:57:08.000000000 -0300
+++ gdb-6.5/bfd/elfcode.h	2006-07-07 02:36:03.000000000 -0300
@@ -1625,7 +1625,8 @@ NAME(_bfd_elf,bfd_from_remote_memory)
 	  if (segment_end > (bfd_vma) contents_size)
 	    contents_size = segment_end;
 
-	  if ((i_phdrs[i].p_offset & -i_phdrs[i].p_align) == 0)
+	  if ((i_phdrs[i].p_offset & -i_phdrs[i].p_align) == 0
+	      && loadbase == ehdr_vma)
 	    loadbase = ehdr_vma - (i_phdrs[i].p_vaddr & -i_phdrs[i].p_align);
 
 	  last_phdr = &i_phdrs[i];
